bin/templates/cordova/Api.js

```diff
@@ -0,0 +1,506 @@
+/**
+    Licensed to the Apache Software Foundation (ASF) under one
+    or more contributor license agreements.  See the NOTICE file
+    distributed with this work for additional information
+    regarding copyright ownership.  The ASF licenses this file
+    to you under the Apache License, Version 2.0 (the
+    "License"); you may not use this file except in compliance
+    with the License.  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing,
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+*/
+
+var Q = require('q');
+var fs = require('fs');
+var path = require('path');
+var shell = require('shelljs');
+
+var CordovaError = require('cordova-common').CordovaError;
+var PlatformJson = require('cordova-common').PlatformJson;
+var ActionStack = require('cordova-common').ActionStack;
+var AndroidProject = require('./lib/AndroidProject');
+var PlatformMunger = require('cordova-common').ConfigChanges.PlatformMunger;
+var PluginInfoProvider = require('cordova-common').PluginInfoProvider;
+
+var pluginHandlers = require('./lib/pluginHandlers');
+
+var PLATFORM = 'android';
+var GENERIC_EVENTS = new (require('events').EventEmitter)()
+    .on('verbose', function (message) {
+        if (process.argv.indexOf('-d') >= 0 || process.argv.indexOf('--verbose') >= 0)
+            console.log(message);
+    })
+    .on('log', console.log)
+    .on('warn', console.warn)
+    .on('error', function (error) {
+        if (process.argv.indexOf('-d') >= 0 || process.argv.indexOf('--verbose') >= 0) {
+            console.error((error && error.message) || error);

```

Comment:
Should this be flipped? We should `stack` for verbose output and not otherwise? Also, does the message not show when we show stack? If I understand correctly, this is only for legacy scenarios - correct? Otherwise, the cli should eventually be doing the logging - correct?